### ISSUE TITLE
[CC] Can't tell if this is supposed to happen... Food particles are totally jacked up

### ISSUE THREAD

**Issue by zzApotheosis** [ReikaKalseki/Reika_Mods_Issues#1237]
64-bit Java 1.8.0_112-b15
Minecraft 1.7.10
MCP v9.05
FML v7.10.99.99
Minecraft Forge 10.13.4.1614
DragonAPI 1.7.10 V16d
ChromatiCraft 1.7.10 V16d

I've known about this bug for a while now, but I just got around to running some tests with the mod and it seems as though I can easily reproduce the bug every time. My private modpack contains 245 mods loaded, 240 mods active, which is a very large pack, so I assumed that one of the mods in the pack was conflicting with ChromatiCraft.

I removed every mod from my pack except for the core mods and libraries, which was around 20 of them. Still the bug was there.

I removed about 10 more, still the bug was there.

So I made a totally fresh instance of Minecraft, with 100% default configs and untouched settings, with only DragonAPI and ChromatiCraft. The barebones minimum. Still the bug was there.

In order to reproduce the bug, just install DragonAPI and ChromatiCraft and eat any food. The particles that show are totally different from the food being consumed, which is so strange because this definitely should have been obvious. And because it should have been obvious, it leads me to wonder if this was actually intended. I read through #452 so I'm guessing this still isn't really intended.

You can check out my [latest log](https://drive.google.com/open?id=0B_wljXPJi507VEZoSFVPOVd1WFE) (from my very large private modpack) and see if there's anything in there that will help. You can see at around 01:03:37 was when I was eating random stuff to test. This bug never caused a crash, so I haven't a crash report.

**Comment by ReikaKalseki** [comment]
Does it resolve itself without ChromatiCraft? I know of this issue, but it has existed since 1.5, *long* before ChromatiCraft existed.

**Comment by zzApotheosis** [comment]
Yes. Even with 239 mods in my pack, food particles are fine. When I add just ChromatiCraft, the bug happens.

I ran some more tests to experiment:
- Case 1: Totally fresh instance of Minecraft; default configs
Mods: DragonAPI
Results: Food particles were perfectly fine. No reportable bugs.

- Case 2: Added ChromatiCraft to the instance of the previous case
Mods: DragonAPI, ChromatiCraft
Results: Food particles were perfectly fine. No reportable bugs. Note this was still on the totally fresh instance of Minecraft.

So this leads me to wonder if there's something else in my major modpack that's causing this issue, which is so weird because the configs and stuff that are related to the 238 mods I removed should have no effect on ChromatiCraft or DragonAPI since those configs' mods are missing.

- Case 3: Started with a fresh instance of my large modpack; Removed every mod except DragonAPI and ChromatiCraft, leaving the rest of the instance intact (config/, hats/, journeymap/, scripts/, options.txt, and optionsof.txt)
Mods: DragonAPI, ChromatiCraft
Results: Food particles were bugged.

- Case 4: Started with the results of the previous case; Deleted the configs folder to let the client generate default configs for DragonAPI and ChromatiCraft
Mods: DragonAPI, ChromatiCraft
Results: Crashes on startup. [Crash report](https://drive.google.com/open?id=0B_wljXPJi507WW5pQ1JNUFM4STQ) asks me to use HashMap for Sound Categories, so I set that to true. This doesn't have anything to do with this bug report, but I'll document it for good measure.

- Case 5: Started with the results of the previous case;
Mods: DragonAPI, ChromatiCraft
Results: Food particles were bugged.

- Case 6: Started with the results of the previous case; I deleted options.txt and optionsof.txt (even though OptiFine wasn't installed) to let the client generate the default options
Mods: DragonAPI, ChromatiCraft
Results: Food particles were bugged.

- Case 7: Started with totally fresh instance of Minecraft, like Case 1; Added only DragonAPI and ChromatiCraft; enabled HashMap for Sound Categories
Mods: DragonAPI, ChromatiCraft
Results: This one might be able to seriously help. I was able to get food particles to appear normally (non-bugged) when a crystal pylon was not in a loaded chunk by reducing the render distance to 2 chunks. When there was a crystal pylon in a loaded chunk, the food particles starting bugging out.
You can check this test out [here](https://drive.google.com/open?id=0B_wljXPJi507eWlfY004YmE4elk). I should also note in Case 2, I used the same world that I generated in Case 1 to run the test, so there were no crystal pylons in view.

**Comment by ReikaKalseki** [comment]
> I was able to get food particles to appear normally (non-bugged) when a crystal pylon was not in a loaded chunk by reducing the render distance to 2 chunks. When there was a crystal pylon in a loaded chunk, the food particles starting bugging out.
You can check this test out here. I should also note in Case 2, I used the same world that I generated in Case 1 to run the test, so there were no crystal pylons in view.

This issue predates pylons by over a year.

**Comment by zzApotheosis** [comment]
Currently, I can't seem to reproduce the bug with anything other than the chunk-loaded pylons. If you believe pylons aren't the issue, then we're back to square one. I'll let you know if I find anything else in testing.

In the grand scheme of things, this really isn't that big of an issue. It's definitely something I can easily cope with, but this may turn some players away.

**Comment by ReikaKalseki** [comment]
This issue started happening in late 1.5.2. ChromatiCraft was created in 1.7.10.

**Comment by Aivech** [comment]
Just as a guess, it could be whatever ChromatiCraft uses to render the pylon sprite.

**Comment by ReikaKalseki** [comment]
And it took effect over a year before such a sprite was implemented?

**Comment by zzApotheosis** [comment]
> I've known about this bug for a while now

Actually, I've known about this bug for about a year. I didn't report it at the time because I thought somebody else would have reported it. Looking back at #452, it looks like they were trying to link FastCraft with the food particle issue, but I tested this without FastCraft, and the bug was present, as stated above:

> Case 3: Started with a fresh instance of my large modpack; Removed every mod except DragonAPI and ChromatiCraft, leaving the rest of the instance intact (config/, hats/, journeymap/, scripts/, options.txt, and optionsof.txt)
Mods: DragonAPI, ChromatiCraft
Results: Food particles were bugged.

**Comment by Aivech** [comment]
You said this started happening in 1.5. Didn't something about the way MC itself handles textures or something like that change at that time?

**Comment by ReikaKalseki** [comment]
No, it was 1.6 that moved to individual textures.

**Comment by Builderb0y** [comment]
If by "individual textures" you mean the ability for resource packs to overwrite one texture at a time instead of all of them at once, and the fact that resource packs no longer have broken textures for all the blocks/items that they haven't specifically made textures for (except for your stuff since you're still using 3 and a half year old code), then that was 1.5, not 1.6.

**Comment by ReikaKalseki** [comment]
Right, 1.6 was the sound system redesign.
